src/archive/rar.rs

```diff
@@ -8,10 +8,22 @@ use crate::{error::Error, list::FileInArchive, utils::logger::info};
 
 /// Unpacks the archive given by `archive_path` into the folder given by `output_folder`.
 /// Assumes that output_folder is empty
-pub fn unpack_archive(archive_path: &Path, output_folder: &Path, quiet: bool) -> crate::Result<usize> {
+pub fn unpack_archive(
+    archive_path: &Path,
+    output_folder: &Path,
+    password: Option<impl AsRef<[u8]>>,
```

Comment:
We are only passing passwords with a single type, so we don't need this to be generic, you can provide `Option<&[u8]>` here directly.